integration/aggregate_documents_compat_test.go

```diff
@@ -1756,6 +1778,151 @@ func TestAggregateCompatProject(t *testing.T) {
 			},
 			resultType: emptyResult,
 		},
+		"SumValue": {
+			pipeline: bson.A{
+				bson.D{{"$project", bson.D{
+					{"sum", bson.D{{"$sum", "$v"}}},
```

Comment:
Do we need to test nested fields? Like `{"sum", bson.D{{"$sum", "$v.foo.bar"}}}`?

Reply:
Sure, I could add couple more tests :hugs:

Reply:
In tests I added more generic `{"sum", bson.D{{"$sum", "$v.foo"}}}` and `{"sum", bson.D{{"$sum", "$v.0.foo"}}}`.

I also tried `{"sum", bson.D{{"$sum", "$v.foo.bar"}}}` and it worked too. I think it's because even though it's used in projection, path is handled within Expression and it has standardized way of handling them. So it appears to work fine. :+1: